emacs/init.sls

```diff
@@ -1,2 +1,111 @@
-emacs:
-  pkg.installed
+{% from 'emacs/settings.sls' import emacs with context %}
+
+{%- if emacs.from_pkg == True %}
+install-emacs:
+  pkg.installed:
+    - pkgs:
+        - emacs
+{% else %}
+
+
+## Pull down a github release of the desired version
+{% with url = 'https://github.com/emacs-mirror/emacs/archive/%s.tar.gz'|format(emacs.name) %}
```

Comment:
Okay last couple things here, do we need to set the url here or can we just pull from the emac settings?